ROADMAP.md

```diff
@@ -0,0 +1,579 @@
+An intermediary step would be to run libp2p as a daemon, providing peer-to-peer
+network capabilities to all processes on a machine.
+
+**Why?** To ship libp2p as the default P2P network API inside future
+operating systems. Actually, taking it one step further, make the libp2p
+stack *THE default* networking stack.
+
+### ♻️ Packet switching
+
+**What?** A packet-switched libp2p is on the pathway to achieving
+decentralisation and seamless connectivity across heterogeneous networks
+across the globe, that are running on different transports and not
+necessarily connected to the Internet directly. Essentially it's a
+*global transit system* for libp2p traffic.
+
+For illustration, a TCP packet sent via an IP-based local LAN in
+the US connected to the Internet backbone (via a router) could travel
+through the libp2p network, taking multiple hops along the way based on
+routing decisions, and traversing various networks over different
+transports. Eventually it'll reach its destination in Nepal: a node in a
+mesh network, that's in turn connected to a community network with
+intermittent Internet connection via satellite.
+
+To build a reality like this entails doing away with the concept of
+Autonomous Systems (AS), and rethinking the Internet protocols and
+structures from the viewpoint of peer-to-peer: BGP, OSPF, peering,
+transit networks, multihoming, IANA, etc.
+
+**Why?** It's a crucial requirement in our visionary goal of
+decentralising networks. WIP.
+
+**Links:**
+
+-   [On \"DialMe\", NAT traversal, Relay, line switching, packet switching
+    (libp2p protocols)](https://github.com/ipfs/notes/issues/110)
+
+## Evolve
+
+**Our short-term roadmap**.
+
+This is the stuff pushing the existing libp2p stack forward.
+
+### 🤝 Low latency, efficient connection handshake
+
+**High priority for: IPFS**
+
+**What?** Establishing a connection and performing the initial handshake
+should be as cheap and fast as possible. Supporting things like
+*selective* stream opening, *preemption*, *speculative* negotiation,
+*upfront* negotiation, protocol table *pinning*, etc. may enable us to
+achieve lower latencies when establishing connections, and even the
+0-RTT holy grail in some cases. These features are being discussed in
+the *Protocol Select* protocol design.
+
+**Why?** Multistream 1.0 is chatty and naïve. Streams are essential to
+libp2p, and negotiating them is currently inefficient in a number of
+scenarios. Also, bootstrapping a multiplexed connection is currently
+guesswork (we test protocols one by one, incurring in significant
+ping-pong).
+
+**Links:**
+
+- [Multiselect 2.0 design doc](https://github.com/libp2p/specs/pull/205).
+
+- [Multiselect 2.0 specification](https://github.com/libp2p/specs/pull/227)
+
+- [RFC Multistream 2.0](https://github.com/libp2p/specs/pull/95).
+
+### 🕸 Unprecedented global connectivity
+
+**What?** A DHT crawl measurements (Nov 22nd 2019) showed that out
```

Comment:
Can we link to where this data was collected?
Also, can this be an automated process that runs periodically (e.g., weekly, monthly) so we can see improvements to this over time?

Reply:
Unfortunately I don't know the source of these crawl measurements. Maybe @vyzo or @aschmahmann know more?

>  Also, can this be an automated process that runs periodically (e.g., weekly, monthly) so we can see improvements to this over time?

I would expect there is already such tool. If not, we might be able to extend the periodically run [ipfs-crawler](https://github.com/wiberlin/ipfs-crawler) project.

:+1: on running this periodically to see our improvements over time.